### PR TITLE
GOOWOO-755: Migrate content.accounts.get to MAPI accounts.get

### PR DESCRIPTION
**Pull request by alaca**
### Changes proposed in this Pull Request:

Closes https://linear.app/a8c/issue/GOOWOO-755/migrate-contentaccountsget-to-mapi-accountsget

Migrates the Merchant Center account reads not covered by the website-claim ticket from the Content API accounts.get to the MAPI

### Detailed test instructions:

- Connect Google for WooCommerce and open the contact-information / store-requirements screen. Confirm the Merchant Center address, phone, and verification status display as before.
- Confirm the "MC address differs from store address" indicator is correct when they match (including the state/region), and that updating the address still works.
- Check the connection status , merchant access still resolves (yes/no) for the connected account.

### Changelog entry
Migrate Merchant Center business info and account access reads to the MAPI.
>

### CODE REVIEW

File: src/API/Google/Merchant.php [hunk 1]
```diff
@@ -341,21 +349,17 @@ public function link_ads_id( int $ads_id ): bool {
...
-			foreach ( $account->getUsers() as $user ) {
-				if ( $email === $user->getEmailAddress() && $user->getAdmin() ) {
-					return true;
-				}
-			}
-		} catch ( GoogleException $e ) {
-			do_action( 'woocommerce_gla_mc_client_exception', $e, __METHOD__ );
+			$user = $this->users_service->get_current_user();
+		} catch ( MerchantApiException $e ) {
+			// The woocommerce_gla_mc_client_exception action is fired by MerchantApiException::__construct().
```
Comment (joemcgill) [inline src/API/Google/Merchant.php:379]: Would there be any reason why we would want to return the error to the caller here rather than simply returning `false`?

File: src/MerchantCenter/ContactInformation.php [hunk 2]
```diff
@@ -48,40 +50,94 @@ public function __construct( Merchant $merchant, Settings $settings ) {
...
+	 * @return AccountBusinessInformation The contact information associated with the Merchant Center account.
 	 *
 	 * @throws ExceptionWithResponseData If the Merchant Center account can't be retrieved or updated.
 	 */
-	protected function update_contact_information( AccountBusinessInformation $business_information ): void {
-		$account = $this->merchant->get_account();
+	public function update_address_based_on_store_settings(): AccountBusinessInformation {
+		// Read the current business information from the account so the address update
+		// preserves the other fields (phone, customer service) instead of the reduced
+		// view returned by get_contact_information().
```
Comment (joemcgill) [inline src/MerchantCenter/ContactInformation.php]: Claude is flagging that this method still calls `get_account()` / Content API:

>`get_contact_information` was fully migrated to MAPI, but the update path still reads via `$this->merchant->get_account()` and writes via `update_account()` to preserve `phone/customer-service fields.` This creates a split-API situation: reading from MAPI, writing through the Content API. If the two resources diverge or the Content API is deprecated first, the update path will silently stomp MAPI-only fields. This should either be migrated in this PR or tracked with a concrete follow-up ticket reference in the comment.

**Review by joemcgill — CHANGES_REQUESTED** [review]
This looks pretty good. I left a few comments below. I also notice that some PHPUnit tests are now failing due to deprecations from WC 10.8. It may be best to mark those as expected deprecations in this PR and open a separate issue in Linear to ensure those paths get removed. Related details can be found in `https://github.com/woocommerce/woocommerce/issues/64444`

File: src/API/Google/Merchant.php (hunk 1 above)
Comment (alaca) [reply src/API/Google/Merchant.php:379]: To clarify, the method does return false here (the line just below this comment) nothing is thrown or returned to the caller.

**Review by joemcgill — COMMENTED** [review]
## 🤖 AI Code Review

**Issue:** [GOOWOO-755 — Migrate content.accounts.get to MAPI accounts.get](https://linear.app/a8c/issue/GOOWOO-755/migrate-contentaccountsget-to-mapi-accountsget)
**Branch:** `feature/GOOWOO-755-accounts-get-mapi` → `feature/mapi-migration`

### Summary

Both acceptance criteria in scope for this PR are satisfied: `ContactInformation::get_contact_information()` now reads from the MAPI `businessInfo` resource via `Merchant::get_business_info()`, and `Merchant::has_access()` now reads from `users/me` via the MAPI users resource. The third criterion (removing `get_account()`) is explicitly deferred pending the Ads-link migration ticket, and `get_account()` still has live callers, so there's no dead code. Overall the implementation is clean and well-tested; the items below are mostly edge-case hardening and a small test gap.

### 🟡 Suggestions

- **Blanket 403 treated as claim conflict** (`src/API/Google/Merchant.php`): `is_website_claim_conflict_error` returns `true` for any MAPI HTTP 403, carried over from the Content API era. On MAPI, 403 is the standard Forbidden/Unauthorized code — a real auth failure during `claim()` would be swallowed and surfaced as a misleading "already claimed" message. Worth a comment on the concrete MAPI scenario this defends, or removing it if unreachable.
- **`map_to_postal_address` sends empty strings for null address fields** (`src/MerchantCenter/ContactInformation.php`): Since the update mask is `'address'` (full replacement), empty-string fields could clear previously-set MAPI sub-fields for stores with incomplete addresses. Consider omitting empty keys.
- **`rawurlencode` on comma-separated update mask** (`MapiAccountBusinessInfoService.php:65`): No active bug since only `'address'` is passed today, but a future multi-field mask would get incorrectly encoded. Consider accepting an array or documenting the constraint.
- **Missing test for `update_address_based_on_store_settings()` exception path**: The read path has `test_get_business_info_exception`; no symmetrical test for the write path's exception handling.
- **`AccountService` resolves `MapiAccountHomepageService` via container->get() instead of constructor injection**, inconsistent with `Merchant`/`ContactInformation`. Pre-existing pattern in this class, not a regression, but worth flagging as consistency debt.

### 🔵 Nitpicks

- `test_claimwebsite_error` (500 case) doesn't assert the exception code, unlike its 400/403 siblings.
- `MapiAccountUsersServiceTest` has only one test case; could use a path-construction test like its sibling services.
- `@throws` docblocks on `claimwebsite()`/`has_access()` could note why `MerchantApiException` doesn't propagate.

---
*This review was generated by Claude. A human review is still required before merging.*

File: src/API/Google/Merchant.php (hunk 1 above)
Comment (joemcgill) [reply src/API/Google/Merchant.php:379]: Apologies, my comment was unclear. I understand that this returns false currently. I'm curious if we should return the exception instead of swallowing this (even though it does get logged). I also notice that the docblock says that this exception gets thrown, but that is inaccurate with the current implementation.

I'd suggest at least making the implementation and the docblocks align here.

**Review by joemcgill — COMMENTED** [review]
The MAPI updates look great and I think this is pretty close. I've pasted another Claude review just for your visibility, but doesn't look like anything critical needs to be addressed. 

A couple of things I would like to see adjusted still:

1. I've left a new inline comment clarifying my `MerchantApiException` exception.
2. The deprecation errors are still causing the PHPUnit tests to fail. Let's ensure that is addressed before merging this work.

**Review by joemcgill — APPROVED** [review]
